collection.go

```diff
@@ -679,6 +679,56 @@ func LoadCollection(file string) (*Collection, error) {
 	return NewCollection(spec)
 }
 
+// Assign the contents of a Collection to a struct.
+//
+// This function bridges functionality between bpf2go generated
+// code and any functionality better implemented in Collection.
+//
+// 'to' must be a pointer to a struct. A field of the
+// struct is updated with values from Programs or Maps if it
+// has an `ebpf` tag and its type is *Program or *Map.
+// The tag's value specifies the name of the program or map as
+// found in the CollectionSpec.
+//
+//	struct {
+//	    Foo     *ebpf.Program `ebpf:"xdp_foo"`
+//	    Bar     *ebpf.Map     `ebpf:"bar_map"`
+//	    Ignored int
+//	}
+//
+// Returns an error if any of the eBPF objects can't be found, or
+// if the same Map or Program is assigned multiple times.
+//
+// Ownership and Close()ing responsibility is transferred to `to`
+// for any successful assigns.
+func (coll *Collection) Assign(to interface{}) error {
+	// Assign() only transfers already-loaded Maps and Programs. No extra
+	// loading is done.
+	getValue := func(typ reflect.Type, name string) (interface{}, error) {
+		switch typ {
+
+		case reflect.TypeOf((*Program)(nil)):
+			if p := coll.Programs[name]; p != nil {
+				delete(coll.Programs, name)
```

Comment:
From the doc:

```
// Ownership and Close()ing responsibility is transferred to `to`
// for any successful assigns.
```

This transfers ownership even if the assign fails?

Reply:
I meant to say partial assigns are possible. Seemed tricky to do rollbacks here. And also `Assign()` being called on something not generated by bpf2go seems like a rare case. So an actual error is unlikely.

WDYT? Update documentation or implement rollbacks?

Reply:
Rollbacks don't have to be complicated I think, see `LoadAndAssign`. If it's okay I'd prefer rollbacks since that keeps semantics consistent.

Reply:
Ack, will take a look